docs/node/workflows.md

```diff
@@ -65,54 +65,81 @@ There is no official support for querying the state of asynchronously started Wo
 
 ## How to cancel a Workflow Execution
 
-In the Node SDK, Workflows are represented internally by a tree of scopes. The main function runs in the root scope.
-Cancellation propagates from outer scopes to inner ones and is handled by catching `CancelledError`s
-thrown by cancellable operations (see below).
+In the Node SDK, Workflows are represented internally by a tree of **Cancellation Scopes**, each with cancellation behaviors you can specify.
+The Workflow `main` function is the only scope with no parent, running as the "root" scope.
 
-Scopes are created using the [`CancellationScope`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope) constructor or the static helper methods
-[`cancellable`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#cancellable-1),
-[`nonCancellable`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#noncancellable),
-and [`withTimeout`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#withtimeout).
+Scopes are created using the [`CancellationScope`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope) constructor, or one of 3 static helpers:
+
+- [`cancellable(fn)`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#cancellable-1) : children are automatically cancelled when their containing scope is cancelled.
+  - Equivalent to `new CancellationScope().run(fn)`.
+- [`nonCancellable(fn)`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#noncancellable): prevents cancellation from propagating to children.
+  - Equivalent to `new CancellationScope({ cancellable: false }).run(fn)`.
+- [`withTimeout(timeoutMs, fn)`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#withtimeout): If timeout triggers before all activities complete, the Workflow will fail with a `CancelledError`.
+  - Equivalent to `new CancellationScope({ cancellable: true, timeout: timeoutMs }).run(fn)`.
+
+Cancellation propagates from outer scopes to inner ones and is handled by catching `CancelledError`s thrown by cancellable operations (see below).
+
+Scopes all return native JS Promises, so you can use all the familiar Promise APIs like `Promise.all` and `Promise.race` to model your async logic.
+Other APIs you can use:
+
+- `CancellationScope.current()`: get the current scope
+- `scope.cancel()`: cancel all operations inside a `scope`
+- `scope.run()`: run an async function within a `scope`
+- `scope.cancelRequested()`: a promise that resolves when a cancel request hits a `nonCancellable` workflow
```

Comment:
I don't know what "hits nonCancellable workflow' means and how it applies to cancelRequested.

I believe `cancelRequested` indicates if the scope was canceled. For example `CancellationScope.current().cancelRequested()` allows to check if the current scope is canceled.

Reply:
```suggestion
- `scope.cancelRequested`: a promise that resolves when a scope cancellation is requested, e.g when Workflow code calls `cancel()` or the entire Workflow is cancelled by an external client.
```